### PR TITLE
updated black version to 20.8b1

### PR DESCRIPTION
**Pull request by zyxue**
updated and reran black on `python/seldon_core` and `python/tests`

discussed on https://github.com/SeldonIO/seldon-core/pull/2713#issuecomment-735827039

Should black be run on all the pb2 files too like `seldon_core/proto/prediction_pb2.py`?

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @zyxue. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/SeldonIO/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **adriangonz**
You can assign the PR to them by writing `/assign @adriangonz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: python/Makefile
Comment (RafalSkolasinski) [inline python/Makefile:98]: Is this a typo?

**Review by RafalSkolasinski — CHANGES_REQUESTED** [review]

**Comment by RafalSkolasinski** [comment]
Nice one!

> Should black be run on all the pb2 files too like seldon_core/proto/prediction_pb2.py?

As these are auto-generated by protos compiler I don't think they should be covered by black.

File: python/Makefile
Comment (RafalSkolasinski) [reply python/Makefile:98]: This should also be updated in `requirements.txt`

Comment (zyxue) [reply python/Makefile]: thanks for the catch. updated

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by zyxue** [comment]
> Nice one!
> 
> > Should black be run on all the pb2 files too like seldon_core/proto/prediction_pb2.py?
> 
> As these are auto-generated by protos compiler I don't think they should be covered by black.

I see. I used to format them for better readability, and also because black won't change the logic in anyway. But not blacking them sounds good too. 

I've addressed your comments.

**Comment by seldondev** [comment]
Wed Dec 16 16:39:25 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=2

**Comment by seldondev** [comment]
Wed Dec 16 16:39:36 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=3

**Comment by seldondev** [comment]
Wed Dec 16 16:39:35 UTC 2020
The logs for [integration] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=4

**Comment by seldondev** [comment]
Wed Dec 16 16:39:37 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=1

**Comment by seldondev** [comment]
@zyxue: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | d718af1e378b048d0d4fe457e7f95de609ab5251 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Wed Dec 16 17:35:25 UTC 2020
The logs for [integration] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=5

**Comment by seldondev** [comment]
@zyxue: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | d718af1e378b048d0d4fe457e7f95de609ab5251 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by zyxue** [comment]
when can I see the log of the failed integration test?

**Comment by RafalSkolasinski** [comment]
there's a link in `seldondev` bot's message, they are available after the pipeline run

**Comment by zyxue** [comment]
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/5.log
```
=================================== FAILURES ===================================
__________________________ TestPrepack.test_tfserving __________________________
[gw1] linux -- Python 3.6.10 /usr/local/bin/python

self = <test_prepackaged_servers.TestPrepack object at 0x7f28bc4376a0>
namespace = 'test-tfserving'

    def test_tfserving(self, namespace):
        spec = "../../servers/tfserving/samples/mnist_rest.yaml"
        retry_run(f"kubectl apply -f {spec}  -n {namespace}")
        wait_for_status("tfserving", namespace)
>       wait_for_rollout("tfserving", namespace)

test_prepackaged_servers.py:97: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

sdep_name = 'tfserving', namespace = 'test-tfserving', attempts = 50, sleep = 5
expected_deployments = 1

    def wait_for_rollout(
        sdep_name, namespace, attempts=50, sleep=5, expected_deployments=1
    ):
        deployment_names = []
        for _ in range(attempts):
            deployment_names = get_deployment_names(sdep_name, namespace)
            deployments = len(deployment_names)
    
            if deployments == expected_deployments:
                break
            time.sleep(sleep)
    
        error_msg = (
            f"Expected {expected_deployments} deployment(s) but got {len(deployment_names)}"
        )
        assert len(deployment_names) == expected_deployments, error_msg
    
        for deployment_name in deployment_names:
            logging.info(f"Waiting for deployment {deployment_name}")
            for _ in range(attempts):
                ret = run(
                    f"kubectl rollout status -n {namespace} deploy/{deployment_name}",
                    shell=True,
                )
                if ret.returncode == 0:
                    logging.info(f"Successfully waited for deployment {deployment_name}")
                    break
                logging.warning(
                    f"Unsuccessful wait command but retrying for {deployment_name}"
                )
                time.sleep(sleep)
            assert (
                ret.returncode == 0
>           ), f"Wait for rollout of {deployment_name} failed: non-zero return code"
E           AssertionError: Wait for rollout of tfserving-default-0-mnist-model failed: non-zero return code

seldon_e2e_utils.py:158: AssertionError
```
this seems have nothing to do with the chage here...

**Comment by cliveseldon** [comment]
Can you rebase as this failure is fixed

**Comment by zyxue** [comment]
rebased. Is test triggered automatically?

**Comment by seldondev** [comment]
Thu Dec 17 16:55:27 UTC 2020
The logs for [pr-build] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=6

**Comment by seldondev** [comment]
Thu Dec 17 16:55:32 UTC 2020
The logs for [lint] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=7

**Comment by seldondev** [comment]
Thu Dec 17 17:04:59 UTC 2020
The logs for [integration] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=8

**Comment by zyxue** [comment]
wondering if there is anything else that needs update for this PR?

**Review by RafalSkolasinski — APPROVED** [review]
Fantastic job!
/lgtm
/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/2769#issuecomment-753933489 "LGTM")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-753933451/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/2769#issuecomment-753933489 "LGTM")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-753933489/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Jan  4 11:56:48 UTC 2021
The logs for [pr-build] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=9

**Comment by seldondev** [comment]
Mon Jan  4 11:56:50 UTC 2021
The logs for [lint] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2769/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2769 --build=10

**Pull request merged**
